syntax="proto3";

option go_package = "github.com/wuisawesome/sky-atc/grpc";

message ConfigureNodeRequest {
  string core_v1_node_json = 1;
}

message ConfigureNodeReply {
  string core_v1_node_json = 1;
}

message CreatePodRequest {
  string core_v1_pod_json = 1;
}

message CreatePodReply {
  // Intentionally empty.
}

message DeletePodRequest {
  string core_v1_pod_json = 1;
}

message DeletePodReply {
  // Intentionally empty.
}

message GetPodsRequest {
}

message GetPodsReply {
  repeated string core_v1_node_jsons = 1;
}

service PodProvider {

  rpc ConfigureNode(ConfigureNodeRequest) returns (ConfigureNodeReply);

  rpc CreatePod(CreatePodRequest) returns (CreatePodReply);

  rpc DeletePod(DeletePodRequest) returns (DeletePodReply);

}
